packages/sdk/tests/test-voucher-utils.ts

```diff
@@ -0,0 +1,388 @@
+    },
+    isAnvil: true,
+  },
+};
+
+const faucetSendWeiToReachTargetBalance =
+  (chain) =>
+  async (address, targetWeiBalance, tryCount = 1) => {
+    const currentBalance = await chain.provider.getBalance(address);
+    const delta = BigInt(`${targetWeiBalance}`) - currentBalance;
+    if (delta < 0n) {
+      console.warn(
+        `Faucet send Eth: aborted - current balance exceed target balance`
+      );
+      return;
+    }
+    try {
+      const tx = await chain.faucetWallet
+        .connect(chain.provider)
+        .sendTransaction({ to: address, value: delta });
+      await tx.wait();
+    } catch (e) {
+      console.warn(`Faucet send Eth: error (try count ${tryCount}) - ${e}`);
+      // retry as concurrent calls can lead to nonce collisions on the faucet wallet
+      if (tryCount < 3) {
+        await sleep(3000 * tryCount);
+        await faucetSendWeiToReachTargetBalance(chain)(
+          address,
+          targetWeiBalance,
+          tryCount + 1
+        );
+      } else {
+        throw Error(`Failed to send Eth from faucet (tried ${tryCount} times)`);
+      }
+    }
+  };
+
+export const setBalance = (chain) => async (address, targetWeiBalance) => {
+  if (chain.isAnvil) {
+    await fetch(chain.rpcURL, {
+      method: 'POST',
+      body: JSON.stringify({
+        method: 'anvil_setBalance',
+        params: [address, ethers.toBeHex(targetWeiBalance)],
+        id: 1,
+        jsonrpc: '2.0',
+      }),
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    });
+  } else {
+    await faucetSendWeiToReachTargetBalance(chain)(address, targetWeiBalance);
+  }
+};
+
+const faucetSendNRlcToReachTargetBalance =
+  (chain) =>
+  async (address, nRlcTargetBalance, tryCount = 1) => {
+    const iexec = new IExec(
+      {
+        ethProvider: getSignerFromPrivateKey(
+          chain.rpcURL,
+          chain.faucetWallet.privateKey
+        ),
+      },
+      { hubAddress: chain.hubAddress }
+    );
+    const { nRLC } = await iexec.wallet.checkBalances(address);
+    const delta = BigInt(`${nRlcTargetBalance}`) - BigInt(`${nRLC}`);
+    if (delta < 0n) {
+      console.warn(
+        `Faucet send RLC: aborted - current balance exceed target balance`
+      );
+      return;
+    }
+    try {
+      await iexec.wallet.sendRLC(`${delta}`, address);
+    } catch (e) {
+      console.warn(`Faucet send RLC: error (try count ${tryCount}) - ${e}`);
+      // retry as concurrent calls can lead to nonce collisions on the faucet wallet
+      if (tryCount < 3) {
+        await sleep(3000 * tryCount);
+        await faucetSendNRlcToReachTargetBalance(chain)(
+          address,
+          nRlcTargetBalance,
+          tryCount + 1
+        );
+      } else {
+        throw Error(`Failed to send RLC from faucet (tried ${tryCount} times)`);
+      }
+    }
+  };
+
+export const setNRlcBalance = (chain) => async (address, nRlcTargetBalance) => {
+  if (chain.isNative || chain.defaults?.isNative) {
```

Comment:
in this repo context, we always use the native asset as the protocol token, this is always true, the if statement can be removed as well as the whole `faucetSendNRlcToReachTargetBalance` implementation.